waspc/src/Wasp/Analyzer/TypeDefinitions/TH/Decl.hs

```diff
@@ -275,12 +275,11 @@ waspKindOfHaskellType typ = do
   maybeCustomEvaluationKind <- tryCastingToCustomEvaluationKind typ
   maybeRecordKind <- tryCastingToRecordKind typ
   maybe (fail $ "No translation to wasp type for type " ++ show typ) return $
-    maybeDeclRefKind
+    -- NOTE: It is important to have @maybeCustomEvaluationKind@ first, since we want it to override
```

Comment:
I actually caught a bug I introduced some time ago here, with the wrong order of these checks, and here I fixed it. I wrote these NOTEs so that bug doesn't get reintroduced again. As I am writing this, I am thinking that it would be even better if I introduced tests that ensure those bugs don't happen again. However they would be somewhat complex so I think they are not worth it at the moment, NOTEs should take care of it well enough.

Reply:
Yeah agree, tests would be nice here, or some other Haskell syntactic construct to enforce the constraint, but NOTEs are probably good enough for now as you noted 👍🏻

Reply:
I am a bit out of my depth here, but sounds good :D